app/request_handlers/yreflection_single_reply.hpp

```diff
@@ -0,0 +1,107 @@
+//
+// request_handler.hpp
+// ~~~~~~~~~~~~~~~~~~~
+//
+// Copyright (c) 2003-2015 Christopher M. Kohlhoff (chris at kohlhoff dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef APP_REQUEST_HANDLERS_YREFLECTION_SINGLE_REPLY_HPP
+#define APP_REQUEST_HANDLERS_YREFLECTION_SINGLE_REPLY_HPP
+
+#include <string>
+
+#include <http/server/reply.hpp>
+#include <http/server/request.hpp>
+
+#include <model/data/mailbox.h>
+#include <model/reflection/message.h>
+#include <yamail/data/serialization/json_writer.h>
+
+namespace http {
+namespace server {
+
+/// The common handler for all incoming requests.
+class request_handler {
+public:
+    request_handler(const request_handler&) = delete;
+    request_handler& operator=(const request_handler&) = delete;
+
+    /// Construct with a directory containing files to be served.
+    request_handler();
+    request_handler(request_handler&& other);
+
+    /// Handle a request and produce a reply.
+    /// OnReply must be function object of type "void(reply)".
+    template<typename OnReply>
+    void handle_request(const request& req, OnReply handler);
+
+private:
+    /// The directory containing the files to be served.
+    using Mailbox = model::Mailbox<model::DummyImpl>;
+    Mailbox mailbox;
+
+    template<typename OnReply>
+    struct reply_collector : Mailbox::Continuation<reply_collector<OnReply>> {
```

Comment:
I think it is not good idea. With such architecture you can not use lambda as continuation and what is pity.